requests/sessions.py

```diff
@@ -222,9 +222,12 @@ def rebuild_proxies(self, prepared_request, proxies):
         url = prepared_request.url
         scheme = urlparse(url).scheme
         new_proxies = proxies.copy() if proxies is not None else {}
+        no_proxy = proxies.get('no_proxy') if proxies is not None else None
```

Comment:
At this point we're doing two `if proxies is not None` checks one after another. Given that the only places we use it in this function are these two lines, let's just add a line at the start that says `proxies = proxies if proxies is not None else {}`. That lets us remove the checks from these two lines.